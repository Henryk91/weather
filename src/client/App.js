
import React, { Component } from 'react'
import { BrowserRouter as Router, Route, Link } from 'react-router-dom';
import { History, Home, Detailed, SearchBar } from './views/Components/index';

export default class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      weatherData: '',
    }
    this.setWeatherData = this.setWeatherData.bind(this)
  }

  setWeatherData(val) {
    this.setState({ weatherData: val.weatherData })
  }
  render() {
    const weatherData = this.state.weatherData;
    return (
      <Router>
        <div>
          <header>
            <SearchBar set={this.setWeatherData} />
            <nav className="bigScreen" id="links">
              <Link className="yellowLink" to="/" title="Weather By Week">This Week</Link>
              <Link className="yellowLink" to="/history" title="Weather By Day">Select By Day</Link>
            </nav>
          </header>
          <Route
            exact path='/'
            render={(props) => <Home {...props} weatherData={weatherData} />}
          />
          <Route
            exact path='/detailed/:id'
            render={(props) => <Detailed {...props} weatherData={weatherData} />}
          />
          <Route exact path="/history"
            render={(props) => <History {...props} weatherData={weatherData} />}
          />
          <Link to="https://darksky.net/poweredby/" target="_blank">
            <img id="darkSkyLogo" src="https://darksky.net/dev/img/attribution/poweredby-oneline-darkbackground.png"
              alt="powered by darksky" />
          </Link>
        </div>
      </Router>
    )
  }
}
