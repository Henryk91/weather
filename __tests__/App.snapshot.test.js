
import App from '../src/client/App';
import renderer from 'react-test-renderer';
import React from 'react';

test('History component renders with detaild and date checker', () => {

  const rendered = renderer.create(
    <App />
  );
  expect(rendered.toJSON()).toMatchSnapshot();
});