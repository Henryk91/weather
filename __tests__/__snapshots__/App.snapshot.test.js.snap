// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History component renders with detaild and date checker 1`] = `
<div>
  <header>
    <header>
      <button
        className="searchButton blueHover"
        id="geoButton"
        onClick={[Function]}
        title="Geo Locate"
      >
        <i
          className="fas fa-map-marker-alt"
        />
      </button>
      <input
        aria-label="Type Location Here"
        id="locationBox"
        placeholder="Type Location..."
        required="required"
        type="text"
      />
      <button
        className="searchButton blueHover"
        id="goButton"
        onClick={[Function]}
        title="Search"
      >
        <i
          className="fas fa-search"
        />
      </button>
    </header>
    <nav
      className="bigScreen"
      id="links"
    >
      <a
        className="yellowLink"
        href="/"
        onClick={[Function]}
        title="Weather By Week"
      >
        This Week
      </a>
      <a
        className="yellowLink"
        href="/history"
        onClick={[Function]}
        title="Weather By Day"
      >
        Select By Day
      </a>
    </nav>
  </header>
  <div
    className="bigScreen"
    id="home"
  >
    <h3>
      Please enter a location
    </h3>
    <main
      className="weekDays"
    >
       
       
    </main>
  </div>
  <a
    href="/https://darksky.net/poweredby/"
    onClick={[Function]}
    target="_blank"
  >
    <img
      alt="powered by darksky"
      id="darkSkyLogo"
      src="https://darksky.net/dev/img/attribution/poweredby-oneline-darkbackground.png"
    />
  </a>
</div>
`;
